util/cosmos/iscn/sign.js

```diff
@@ -179,17 +182,47 @@ export async function estimateISCNTxFee(tx, {
   return { iscnFee };
 }
 
-export async function estimateISCNTxGas() {
+export async function estimateISCNTxGas(tx) {
+  const record = await formatISCNPayload(tx);
+  const msg = {
+    type: Buffer.from('likecoin-chain/MsgCreateIscnRecord', 'utf-8'),
+    value: {
+      from: Buffer.from(tx.cosmosWallet, 'utf-8'),
+      record,
+    },
+  };
+  // real gasUsed will be slightly diff from gasUsedEstimation
+  // because the value passed into the obj is different from estimation
+  const value = {
+    msg: [msg],
+    fee: {
+      amount: [
+        {
+          denom: 'nanolike',
+          amount: '200000', // temp number here for estimation
+        },
+      ],
+      gas: '200000', // temp number here for estimation
+    },
+  };
+  const obj = {
+    type: 'cosmos-sdk/StdTx',
+    value: Buffer.from(jsonStringify(value), 'utf-8'),
+  };
+  const interceptWithBuffer = BigNumber(GAS_ESTIMATOR_INTERCEPT).plus(GAS_ESTIMATOR_BUFFER);
+  const txBytes = Buffer.from(jsonStringify(obj), 'utf-8');
+  const byteSize = BigNumber(txBytes.length);
+  const gasUsedEstimation = byteSize.multipliedBy(GAS_ESTIMATOR_SLOP).plus(interceptWithBuffer);
   return {
     gasFee: {
-      amount: [{ amount: (DEFAULT_GAS_PRICE_NUMBER * ISCN_GAS).toFixed(), denom: 'nanolike' }],
-      gas: ISCN_GAS.toFixed(),
+      amount: [{ amount: parseInt(gasUsedEstimation.toFixed(0), 10), denom: 'nanolike' }],
+      gas: gasUsedEstimation.toFixed(0),
```

Comment:
```suggestion
      amount: [{ amount: gasUsedEstimation.toFixed(), denom: 'nanolike' }],
      gas: gasUsedEstimation.toFixed(),
```
Should return bignumber or string for nanolike for accuracy, string here is better for consistency